src/Nethermind/Nethermind.Core2.Abstractions/Types/Bytes32.cs

```diff
@@ -31,6 +31,17 @@ public Bytes32()
             _bytes = new byte[Length];
         }
 
+        public Bytes32(byte[] bytes)
```

Comment:
As this is using a reference to the array (rather than a copy), need to be careful about mutating internal state:

var x = new byte[3];
var b1 = new Bytes32(x);
x[1] = (byte)0x01;

Maybe have a private construction and some different name static method, e.g. Bytes.CreateUsing(byte[] bytes) that gives a hint the value is used internally (not copied).

Reply:
Can do a different API - I was worried about the same in Eth1 but in the end it never cause trouble. But probably it is just a metter of time. I will go for CreateUsing or Wrap (the latter clearly indicates that we wrap the byte[] instance)